src/Analyzers/Core/CodeFixes/ConflictMarkerResolution/AbstractConflictMarkerCodeFixProvider.cs

```diff
@@ -45,12 +45,15 @@ internal abstract partial class AbstractResolveConflictMarkerCodeFixProvider : C
         {
             FixableDiagnosticIds = ImmutableArray.Create(diagnosticId);
             _syntaxKinds = syntaxKinds;
+
+#if !CODE_STYLE
+            // Backdoor that allows this provider to use the high-priority bucket.
+            this.CustomTags = this.CustomTags.Add(CodeActionRequestPriorityExtensions.CanBeHighPriorityTag);
+#endif
```

Comment:
how we allow for only our own internal providers to use the 'high pri' bucket for now.

Reply:
both the fix provider, and the fixes have to be marked 'high pri'.

Reply:
Anyone who retursn .High pri from ComputeRequestPriority or who has a fix with .High pri needs to do soemthing like this.  You'll see that in 4 features across thsi PR.

Reply:
Why does this needs to be wrapped around `#if !CODE_STYLE`? I think we should allow even the code style layer to be able to specify high priority providers/code actions, otherwise adding a NuGet package reference to CodeStyle package would downgrade these fixers to normal priority. I thought the primary reason to move to custom tags to specify high priority was to allow third party providers to participate in this. As we don't want anything outside CodeStyle package to specify high priority for time being, we can add additional assembly name guards for the same.

Reply:
> both the fix provider, and the fixes have to be marked 'high pri'.

Why is this required for both providers and code actions?

Reply:
> > both the fix provider, and the fixes have to be marked 'high pri'.
> 
> Why is this required for both providers and code actions?

I see that we are requiring providers to have this custom tag for the clamping support. I can't recall our exact discussion about this in past, but I still feel CodeStyle Nuget package should be able to supply high priority code actions. Which means we need a couple of things:
1. A well-known public API on CodeAction to specify priority, including high priority and
2. A backdoor, not well-known or documented, but still _public_ way to specify "please don't clamp me". This can be done via exposing `CustomTags` as a public property on either the code fix provider or code action (preferably latter). If we don't like this approach, then instead have a hard coded assembly name based check to decide whether or not to clamp.

Reply:
Added a comment https://github.com/dotnet/roslyn/pull/68511/files?diff=unified&w=1#r1264893170, which may be a good middle ground here, without requiring any additional public API or assembly name checks for clamping support.

Reply:
Because a slow 3rd party high PRI provider would be detrimental to all our 1st party providers.  This was the whole idea of keeping that to us for the time being.